app/routes/user/password.tsx

```diff
@@ -0,0 +1,221 @@
+/*!
+ * Copyright © 2022 United States Government as represented by the Administrator
+ * of the National Aeronautics and Space Administration. No copyright is claimed
+ * in the United States under Title 17, U.S. Code. All Other Rights Reserved.
+ *
+ * SPDX-License-Identifier: NASA-1.3
+ */
+import type { DataFunctionArgs } from '@remix-run/node'
+import { useFetcher } from '@remix-run/react'
+import {
+  Alert,
+  Button,
+  ButtonGroup,
+  Icon,
+  InputGroup,
+  Label,
+  TextInput,
+} from '@trussworks/react-uswds'
+import { useEffect, useRef, useState } from 'react'
+import invariant from 'tiny-invariant'
+
+import { getUser } from '../__auth/user.server'
+import { updatePassword } from './password.server'
+import PasswordValidationStep from '~/components/PasswordValidationStep'
+import Spinner from '~/components/Spinner'
+import { getFormDataString } from '~/lib/utils'
+import { useUserIdp } from '~/root'
+
+export const handle = {
+  breadcrumb: 'Reset Password',
+  getSitemapEntries: () => null,
+}
+
+export async function loader({ request }: DataFunctionArgs) {
+  const user = await getUser(request)
+  if (!user) throw new Response(null, { status: 403 })
+  return null
+}
+
+export async function action({ request }: DataFunctionArgs) {
+  const data = await request.formData()
+  const oldPassword = getFormDataString(data, 'oldPassword')
+  const newPassword = getFormDataString(data, 'newPassword')
+
+  invariant(oldPassword && newPassword)
+  return await updatePassword(request, oldPassword, newPassword)
+}
+
+export function WrongIdp() {
+  return (
+    <Alert type="error" heading="Error:" headingLevel="h4">
+      You must be signed in with a username and password to reset your password.
+    </Alert>
+  )
+}
+
+export function ResetPassword() {
+  const fetcher = useFetcher<typeof action>()
+  const errorMessage = fetcher.data
+
+  const [newPassword, setNewPassword] = useState('')
+  const [isOldPasswordTouched, setIsOldPasswordTouched] = useState(false)
+  const [isNewPasswordTouched, setIsNewPasswordTouched] = useState(false)
+  const [visible, setVisible] = useState(false)
+
+  const containsLower = /[a-z]/.test(newPassword)
+  const containsUpper = /[A-Z]/.test(newPassword)
+  const containsNumber = /[0-9]/.test(newPassword)
+  const validLength = newPassword.length >= 8
+  const whitespaceValid = !/^\s|\s$/.test(newPassword)
+  const containsSpecialChar = /[~`!@#$%^.&*+=\-_ [\]\\';,/{}()|\\":<>?]/g.test(
+    newPassword
+  )
+
+  const oldPasswordError = errorMessage && !isOldPasswordTouched
+  const valid =
+    whitespaceValid &&
+    containsLower &&
+    containsUpper &&
+    containsNumber &&
+    validLength &&
+    containsSpecialChar
+  const shouldDisableSubmit =
+    !newPassword || !isNewPasswordTouched || !valid || !isOldPasswordTouched
```

Comment:
If the new password is valid, then it is non-empty and it has been touched.

```suggestion
  const shouldDisableSubmit = !valid || !isOldPasswordTouched
```

Reply:
I'll meet in the middle here. I took out the isNewPasswordTouched check but I left the !newPassword check in because when I don't, on firefox I get the following error. As long as the new password check is included it is happy. 
<img width="822" alt="Screen Shot 2023-07-18 at 12 19 33 PM" src="https://github.com/nasa-gcn/gcn.nasa.gov/assets/6545874/4703484b-c3f2-485d-8556-cbce0fd110e0">